paraatm/simulation_method/aviationr.py

```diff
@@ -0,0 +1,219 @@
+            'model_file' : srt
+                directory to pre-trained model
+            'sim_time' : int/float
+                total simulation duration
+        """
+        self.fp_file = cfg['fp_file']
+        self.mfl_file = cfg['mfl_file']
+        self.data = cfg['data_file']
+        self.case = self.data + 'case.pickle'
+        self.refer = self.data + 'refer.pickle'
+        self.model = cfg['model_file']
+        self.sim_time = cfg['sim_time']
+
+
+        NatsEnvironment.start_jvm(nats_home=None)
+        self.NATS_SIMULATION_STATUS_PAUSE = NatsEnvironment.get_nats_constant('NATS_SIMULATION_STATUS_PAUSE')
+        self.NATS_SIMULATION_STATUS_ENDED = NatsEnvironment.get_nats_constant('NATS_SIMULATION_STATUS_ENDED')
+
+        natsStandalone = NatsEnvironment.get_nats_standalone()
+
+        self.simulationInterface = natsStandalone.getSimulationInterface()
+
+        self.entityInterface = natsStandalone.getEntityInterface()
+        self.controllerInterface = self.entityInterface.getControllerInterface()
+        self.pilotInterface = self.entityInterface.getPilotInterface()
+
+        self.environmentInterface = natsStandalone.getEnvironmentInterface()
+
+        self.equipmentInterface = natsStandalone.getEquipmentInterface()
+        self.aircraftInterface = self.equipmentInterface.getAircraftInterface()
+
+        if self.simulationInterface is None:
+            natsStandalone.stop()
+            raise RuntimeError("Can't get SimulationInterface")
+
+        self.simulationInterface.clear_trajectory()
+
+    def simulation(self, device=None, delay=100, isRNN=True):
+        """
+        Main function for simulation. NATS simulation based commands
+
+        Parameters
+        ----------
+        device : torch.device
+           'CUDA'/ 'CPU'-- Running the model along GPU or CPU
+        delay : int
+            When we start the accident simulation after the required condition meet
+        isRNN : boolean
+            Using RNN or sequential model
+
+        Returns
+        -------
+        dictionary
+            Recorded risk quantification:
+        """
+        # Load data
+        case = pd.read_pickle(self.data + 'case.pickle')
+        case_code = case['case_code']
+        case = case['case']
+        refer = pd.read_pickle(self.data + 'refer.pickle')
+        ntsb2nats = refer['ntsb2nats']
+        datadict = refer['datadict']
+
+        self.simulationInterface.clear_trajectory()
+
+        self.environmentInterface.load_rap(os.environ.get('NATS_HOME') + '/' +"share/tg/rap")
+
+        self.aircraftInterface.load_aircraft(self.fp_file, self.mfl_file)
+
+        self.simulationInterface.setupSimulation(self.sim_time, 30)  # SFO - PHX
+
+        self.simulationInterface.start(1)
+        while True:
+            runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+            if (runtime_sim_status == self.NATS_SIMULATION_STATUS_PAUSE):
+                break
```

Comment:
I reviewed this with Erin.  We think that starting for 1 second and then pausing is not necessary.  You should be able to just call `start()` with no time argument, and then remove this loop to check for a pause, since you are not doing anything with the paused state.
```suggestion
        self.simulationInterface.start()
```

Reply:
Actually, after looking at this some more, I can see that maybe the reason for this initial pause is so that you can have the `resume` call at the start of the `while not accident` loop.  Otherwise, if you remove the initial pause, then `resume` would be called while the filght is not paused, so not 100% sure what that would do.  I suppose you could remove the initial pause and move `resume` somewhere else, such as in an `else` branch on `if accident_phase`, but maybe the way you have it is fine.  One thing we were wondering about is whether starting for 1 second but using a time step of 30 seconds might cause any issues with NATS.